server/routes/api/course.js

```diff
@@ -0,0 +1,223 @@
+                success: false,
+                message: "Error: Server Error"
+            });
+        }
+        if (previousCourse.length > 0){
+            return res.status(409).send({
+                success: false,
+                message: "Error: Course already exists"
+            });
+        }
+    // save the course
+        const newCourse = new Course();
+
+        newCourse.name = name;
+        newCourse.code = code;
+        newCourse.department = department;
+        newCourse.description = description;
+        newCourse.resourcesUrl = resourcesUrl;
+        newCourse.duration.startDate = startDate;
+        newCourse.duration.endDate = endDate;
+        newCourse.details.credits = credits;
+        newCourse.details.hours = hours;
+        newCourse.professors = prof;
+        newCourse.students = students;
+        console.log(newCourse)
+
+        newCourse.save((err, course) =>{
+            if (err) {
+                return res.status(500).send({
+                    success: false,
+                    message: "Error: Server1 error"
+                });
+            }
+            console.log(newCourse._id + " Added to DB")
+            return res.status(200).send({
+                success: true,
+                message: "New course created"
+            })
+        })
+    }
+)
+})
+// end of createCourse endpoint
+app.post('/api/assignments/createAssignment', requireRole("admin"), function(req, res){
+
+    var name = '' + req.body.name;
+    var user = '' + req.body.user;
+    var uniqueId = '' + req.body.uniqueId;
+    var courseCode = '' + req.body.courseCode;
+    var type = '' + req.body.type;
+    var maxMarks = '' + req.body.maxMarks;
+    var resourcesUrl = '' + req.body.resourcesUrl;
+    var endDate = '' + req.body.endDate;
+    var startDate = '' + req.body.startDate;
+    var POC = '' + req.body.POC;
+    var submissionsUrl = '' + req.body.submissionsUrl;
+    console.log(courseCode)
+
+    if (!courseCode || !type || !uniqueId || !name || !submissionsUrl || !user){
+        return res.status(400).send({
+            success: false,
+            message: 'Error: All details are required'
+        });
+    }
+    var courseid = courseCode
+    Assignment.find({
+        courseid: courseid._id,
+        isDeleted: false
+    }, (err, course) => {
+        if (err){
+            return res.status(500).send({
+                success: false,
+                message: "Error: Server1 Error"
+            });
+        }
+    // save the assignments
+    const newAss = new Assignment();
+
+    newAss.name = name;
+    newAss.course = courseCode;
+    newAss.type = type;
+    newAss.submissions.user = user._id;
+    newAss.maxMarks = maxMarks;
+    newAss.resourcesUrl = resourcesUrl;
+    newAss.duration.startDate = startDate;
+    newAss.duration.endDate = endDate;
+    newAss.POC = POC;
+    newAss.uniqueId = uniqueId;
+    newAss.submissionsUrl = submissionsUrl;
+

```

Comment:
Again, check the schema. Some of these are Schema.Types.ObjectID types. How can I send that with a request? You need to do more finds. I shall send the USN.